### PR TITLE
move dataref at beginning of responses so we can allocate less

### PR DESCRIPTION
**Pull request by dbentley**

### CODE REVIEW

File: memory.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ const maxWrite = 16 * 1024 * 1024
 // TODO(dbentley): Why does this seem reasonable?
 const maxReadSize = maxWrite
 
+const maxFilenameLen = 1024 // Maximum filename length we support
```
Comment (grimreaper) [inline memory.go:15]: can this be taken from the OS / libc instead of using our own?

File: memory.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ const maxWrite = 16 * 1024 * 1024
 // TODO(dbentley): Why does this seem reasonable?
 const maxReadSize = maxWrite
```
Comment (grimreaper) [inline memory.go:13]: these are two independent values IMHO

Comment (jasonbs10) [inline memory.go]: lgtm

**Comment by jasonbs10** [comment]
lgtm

**Pull request merged**